### DISCUSSION TITLE
I made an arduino cnc foam cutter for my kid but something went wrong, can anyone fix it?

### DISCUSSION THREAD

**Discussion by huangdajin** [winder/Universal-G-Code-Sender#2587] [Help]
Return to the position failed, only Z axis rotation, X, Y axis no linkage rotation, controller completed the unexpected command
Its provenance comes [from](https://howtomechatronics.com/projects/arduino-cnc-foam-cutting-machine/#google_vignette)
![QQ图片20240805231712](https://github.com/user-attachments/assets/214c5d3b-bb98-4a62-b3b7-cd8bcf6cbd47)
![QQ图片20240805231728](https://github.com/user-attachments/assets/0859e83e-f65d-4d19-8521-30071e260ecb)

**Comment by breiler** [comment]
Are you sure that your limit switches are working?
Did you change the homing sequence in the GRBL config.h?
Have you tried doing what was suggested in the error message, increase the homing pull-off (https://github.com/gnea/grbl/wiki/Grbl-v1.1-Configuration#27---homing-pull-off-mm)?

**Reply by huangdajin** [reply]
Hello, my limit switch is working fine. May I ask where the homing sequence in GRBL config.h was changed?

**Reply by breiler** [reply]
https://github.com/gnea/grbl/blob/master/grbl/config.h#L105

**Reply by huangdajin** [reply]
Gbur configer. h, I changed it step by step.But the problem is the same.Increased homing pull-off How to increase.My USG output $27 is 1.000 (homing switch pull away distance, mm)

**Reply by huangdajin** [reply]
I don't know how to increase homing pull-off

**Reply by breiler** [reply]
Type this in the console `$27=2` which will increase this setting to 2mm.

**Reply by Coldwinternow** [reply]
Where should I modify, I'm not sure, can you teach me that?

**Reply by Coldwinternow** [reply]
这是我的UGS
<img width="604" alt="屏幕截图 2024-08-17 221304" src="https://github.com/user-attachments/assets/093c782e-01e8-4e1f-9338-8eda18ad1145">
<img width="1280" alt="屏幕截图 2024-08-17 220021" src="https://github.com/user-attachments/assets/c4d36810-4801-44f7-bfea-666adea8df11">

**Reply by Coldwinternow** [reply]
I changed the $27 in UGS to$27=50,But after changing it, there are still these prompts as shown in the figure, x, y, and axis still do not move, how should I solve these problems?
<img width="888" alt="屏幕截图 2024-08-17 224647" src="https://github.com/user-attachments/assets/7685d8bd-3ccb-4895-9e6d-a870552699aa">

**Reply by breiler** [reply]
50mm pulloff is a lot so this is another problem. 

Without knowing exactly how your machine is put together, how the controller is wired and how the compile time settings were made it is hard for me to help. 

Since it is an two axis machine you need to have make sure you have disabled the homing sequence for the Z-axis. When you initiate the homing does any of the axes move?

**Reply by huangdajin** [reply]
Why is there only Z-axis movement after home, when the positive limit, X, Y can not automatically return to the origin, the operation code is also based on the video, the wiring is also, the compilation time, the video has, it is set, no, it is not set, in UGS we found that the data is not open the limit setting, but we don't know where the specific location is in this software
![E94F496735FC790FBB42C0DFE6496B3D](https://github.com/user-attachments/assets/92a0bafe-ea2f-448e-ab73-33e8c2d4ccfb)

**Reply by breiler** [reply]
This is impossible for me to answer. If you did everything as the instructions you linked to, I don't see the reason for it not to work. So there is a step that you are missing.

And unless you are able to show me proof that you did everything according to the instructions I am working on guesses. Post pictures of your wiring, screenshot/upload the altered config.h, a video of how the machine is behaving during the homing procedure and videos of how it behaves when trying to operate the machine.

>  the video has, it is set, no, it is not set, in UGS we found that the data is not open the limit setting, but we don't know where the specific location is in this software

I have no idea what you are asking
